Framework/PythonInterface/mantid/__init__.py

```diff
@@ -45,14 +46,37 @@ def _bin_dirs():
     Generate a list of possible paths that contain the Mantid.properties file
     """
     _moduledir = os.path.dirname(os.path.abspath(os.path.dirname(__file__)))
-    # std install
+
+    # standard packaged install
     yield _moduledir
+
     # conda layout
     yield os.path.dirname(sys.executable)
 
+    # The correct path for when the working directory is the same as the
+    # build root directory. Used to run from PyCharm
+    yield sys.path[0]
+
+    try:
+        from mantid.buildconfig import BUILD_DIR
+    except ImportError:
+        # The buildconfig file not found, the function returns to stop the execution of following yields
+        # this will make the _bindir empty check fail later
+        return
 
-# Bail out early if a Mantid.properties files is not found in the
-# 1 of the expected places - it indicates a broken installation or build.
+    # The path for running a development build outside of the working directory
+    yield "{}/bin".format(BUILD_DIR)
+
+    # This path is for when tests are ran. The CMAKE_CONFIG_TYPE is declared from
+    # CTest's -C flag, and it will contain the build type (Debug, Release).
+    # If no config is declared (Linux), then it will be an empty string, and
+    # default to bin/, which is the correct path
+    build_type = os.environ.get('CMAKE_CONFIG_TYPE', '')
```

Comment:
I'm not sure this covers all of the cases on Windows? If I just run `bin\Release\mantidpython`, for example, I would not have the `CMAKE_CONFIG_TYPE` variable defined and I assume this would fail?

Reply:
Perhaps we should just search the `sys.path` and then we wouldn't need `BUILD_DIR` at all?